main.py

```diff
@@ -186,67 +186,91 @@ async def set_playing(status):
     await client.change_presence(activity=discord.Game(name=status))
 
 
-async def give_random(channel, number_of_issues):
-    """
-    @TODO: improve algorithm to save queries
-    1. get first batch of github_max (100), find total number
-    2. if total < requested, return all
-    3. Generate requested random numbers
-    4. get issues till max(generated_list)
-    5. return them
-    """
-    await set_playing('Finding Issues')
-    error_channel = error_handler(channel)
-    async with channel.typing():
-        issues = [x async for x in get_issues(config["repo"],
-                                              exclude_labels=['status: stale'],
-                                              error_channel=error_channel)]
-        reply = discord.Embed(color=discord.Color.purple())
-        reply.title = f"{number_of_issues} random picks out of {len(issues)}:"
-
-        chosen_issues = random.choices(issues, k=number_of_issues)
-        reply.description = compose_message(beautify_issues(chosen_issues))
-        if len(chosen_issues) < number_of_issues:
-            reply.set_footer(text="There wasn't enough...")
-    await channel.send(embed=reply)
-    await set_playing('The Waiting Game')
-
-
-async def review_q(channel, number_of_issues, author=None):
-    await set_playing('On The Cue')
-    error_channel = error_handler(channel)
-    async with channel.typing():
-        issues = get_issues(config["repo"], pull_request=True,
-                            exclude_labels=['status: stale'],
-                            include_labels=['needs: code review'],
-                            sort='updated', ascending_order=True,
-                            error_channel=error_channel)
-        reply = discord.Embed(color=discord.Color.purple())
-
+class SendReply:
```

Comment:
Try to have something like:
```py
# derive from a nice class
class Bot(discord.bot):
    def __init__(self):
        config = read_config(file_name)
        commands = [name: make_command(args) for name, args in config.iteritems()]
    def run(name, value):
        command = commands.get(user_input, lambda x: None) # or throw error, etc.
        command(value)
```

Reply:
Give me some time, I'll rethink the whole structure

Reply:
Would it be better to use [discord.ext](https://discordpy.readthedocs.io/en/latest/ext/commands/)?

Reply:
Yeah, my example and the examples there look similar

Reply:
I was planning on treating commands as [extensions](https://discordpy.readthedocs.io/en/latest/ext/commands/extensions.html#extensions). 
A separate extension for each command or all-in-one?

PS: By commands, I mean feedback queue, review queue etc.

Reply:
Just commands.command decoration might do the trick, without needing the breadth of extensions and cogs.

Reply:
Yes, but it becomes neat and arranged that way. 
Plus, the LOC reduces in the main file (I was planning to move that code to a new file since it is an extension).

Reply:
I have a doubt. What exactly is `error_handler` doing?
https://github.com/PointCloudLibrary/discord-bot/blob/c27394669f1564df397091286532713262f64be0/main.py#L73-L84

While using [exceptions](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#exceptions), I think it will be safe to remove it (or convert it):
```
@bot.on_command_error
async def error_handler(ctx, error):
    reply = discord.Embed(color=discord.Color.purple())
    reply.description = "Talking to me? Use `!what` to know more."
    if isinstance(error, discord.DiscordException):
        await ctx.send(embed=reply)
```

Reply:
> What exactly is `error_handler` doing?

Timeout, API limit hit issues

Reply:
Let's see the prototype and then discuss over it.

Reply:
There might be a problem. We shouldn't share variables in async methods, right? Thus, the whole class thing shouldn't be done. If we really needed that, then we need to make methods synchronous, which involves semaphores, etc. (state variables). (async is threaded afaik)

If that's the case, your original code be should be left as it is, i.e. functional in nature and just use discord.ext to clean some message stuff, make commands, etc.

> Let's see the prototype and then discuss over it.

Sorry for the delay, but the above doubt needs clearing.

Reply:
https://github.com/Rapptz/discord.py/blob/master/examples/basic_voice.py